src/Phobos.Kafka/src/Petabridge.Phobos.Kafka.Consumer/AkkaService.cs

```diff
@@ -162,29 +163,35 @@ private async Task Business(ConsumeResult<Null, string> record)
                 record.Message.Value);
 
             var message = record.Message;
+            var response = await _actors.ConsoleActor.Ask<string>($"hit from {message.Value}", TimeSpan.FromSeconds(5));
+            _logger.LogInformation("[Consumer] Child response: [{ConsumerChildResponse}]", response);
+        }
+    }
 
-            IScope currentScope = null;
-            if (message.Headers.TryGetLastBytes("spanContext", out var contextPayload))
+    public static class KafkaExtensions
+    {
+        public static Source<CommittableMessage<TKey, TValue>, IControl> WithTracing<TKey, TValue>(
```

Comment:
Consumer is simplified by refactoring the span context hydration process out into an extension method.